compiler/semfold.nim

```diff
@@ -490,6 +490,9 @@ proc foldFieldAccess(m: PSym, n: PNode; g: ModuleGraph): PNode =
   var x = getConstExpr(m, n.sons[0], g)
   if x == nil or x.kind notin {nkObjConstr, nkPar, nkTupleConstr}: return
 
+  # this may be a method call in disguise
```

Comment:
Why would the compiler attempt to const fold something that is not yet sem'checked completely?

Reply:
Because when the type() is parsed`transformCall` is invoked on it and `getMagic` returns `mTypeOf`. If you take the code out of the template then the magic field is `mNone`.